### PR TITLE
feat(linux): Add debian packaging GHA

### PR DESCRIPTION
**Pull request by ermshiperete**
There need to be some secrets set and a `deploy` environment set up before this can be merged - as stated in email, or in general in the [doc](https://github.com/sillsdev/gha-ubuntu-packaging/raw/main/PACKAGING.md).

The `deb-packaging` GHA will be triggered by TC, similar to our existing jobs. There's a slight disadvantage in that it's not possible to see from the actions list which workflow run build which branch/PR.

Currently it's not possible to upload to llso with this GHA because we can't access the upload target from the GHA runner. But even without the will this PR improve the availability of packages for testing because they will be available without requiring an account on Jenkins.

Until the upload problem is solved we will still need the Jenkins job.

To see how the GHA would look, see [here](https://github.com/ermshiperete/keyman/actions/runs/3715841238) for a release build, and [here](https://github.com/ermshiperete/keyman/actions/runs/3715851043) for a PR build.

@keymanapp-test-bot skip

### CODE REVIEW

File: .github/workflows/deb-packaging.yml [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+    - name: Install dependencies
+      run: |
+        cd linux
+        sudo mk-build-deps --install --tool='apt-get -o Debug::pkgProblemResolver=yes --no-install-recommends --yes' debian/control
+
+    - name: Build source package
+      run: |
+        cd linux
+        tier="stable"
```
Comment (darcywong00) [inline .github/workflows/deb-packaging.yml:32]: So the packages always use stable tier?

Comment (ermshiperete) [reply .github/workflows/deb-packaging.yml:32]: Good catch!

File: .github/workflows/deb-packaging.yml [hunk 2]
```diff
@@ -0,0 +1,271 @@
+name: "Ubuntu packaging"
+on:
+  repository_dispatch:
+    types: ['deb-release-packaging', 'deb-pr-packaging']
+env:
+  COLOR_GREEN: "\e[32m"
+  GH_TOKEN: ${{ github.token }}
+  STATUS_CONTEXT: 'Debian Packaging'
+  DEBFULLNAME: 'Keyman GHA packager'
+  DEBEMAIL: 'support@keyman.com'
```
Comment (srl295) [inline .github/workflows/deb-packaging.yml:10]: should this go in a secret?

File: .github/workflows/deb-packaging.yml [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+    - name: Download Artifacts
+      uses: actions/download-artifact@9782bd6a9848b53b110e712e20e42d89988822b7 # v3.0.1
+      with:
+        path: artifacts
+
+    - name: Build
+      uses: sillsdev/gha-ubuntu-packaging@5b56c86ee78598308d864edc55a1d06b4ba36f92 # v0.3
+      with:
+        dist: "${{ matrix.dist }}"
+        platform: "${{ matrix.arch }}"
```
Comment (srl295) [inline .github/workflows/deb-packaging.yml:133]: ok, the build itself is matrixed, great

**Review by srl295 — COMMENTED** [review]
+1 by me but i dont' know all of the build issues.

**Comment by ermshiperete** [comment]
Converting to draft until @mcdurdin's back and can add the necessary secrets etc.

**Comment by mcdurdin** [comment]
iOS build failed due to expired certificates, see #8176.

File: .github/workflows/deb-packaging.yml [hunk 4]
```diff
@@ -0,0 +1,271 @@
+name: "Ubuntu packaging"
```
Comment (mcdurdin) [inline .github/workflows/deb-packaging.yml:1]: This file is very heavy with logic. Would it make sense to move more of it to a .sh script?

File: resources/build/run-required-test-builds.sh [hunk 5]
```diff
@@ -58,6 +58,10 @@ function triggerTestBuilds() {
         local job=${test_build%_Jenkins}
         echo "  -- Triggering build configuration $job/$branch on Jenkins"
         triggerJenkinsBuild "$job" "$branch" "$force"
+      elif [ "${test_build:(-7)}" == "_GitHub" ]; then
+        local job=${test_build%_GitHub}
+        echo "  -- Triggering GitHub action build $job/$branch"
+        triggerGitHubActionsBuild true "$job" "$branch"
```
Comment (mcdurdin) [inline resources/build/run-required-test-builds.sh:64]: In due course, we are going to want to track these on status.keyman.com, probably.

File: resources/build/trigger-builds.inc.sh [hunk 6]
```diff
@@ -120,3 +124,39 @@ function triggerJenkinsBuild() {
...
+  local IS_TEST_BUILD="$1"
+  local GITHUB_ACTION="$2"
+  local GIT_BRANCH="${3:-master}"
+  local GIT_REF=$GIT_BRANCH
+
+  local GITHUB_SERVER=https://api.github.com/repos/keymanapp/keyman/dispatches
+
+  # This will only be true if we created and pushed a tag
+  if [ "${action:-""}" == "commit" ]; then
+    GIT_REF="release-$VERSION_WITH_TAG"
```
Comment (mcdurdin) [inline resources/build/trigger-builds.inc.sh:138]:
```suggestion
    GIT_REF="release@$VERSION_WITH_TAG"
```

We've moved to `release@tag` now.

**Review by mcdurdin — COMMENTED** [review]
LGTM, AFAICT. deb-packaging.yml is very heavy, wonder about moving the script portions into a .sh wih appropriate parameters? Not sure it'll help much though.

File: .github/workflows/deb-packaging.yml (hunk 4 above)
Comment (ermshiperete) [reply .github/workflows/deb-packaging.yml:1]: Moved some parts to a `linux/scripts/deb-packaging.sh` script.

**Review by mcdurdin — APPROVED** [review]
LGTM at this point. I feel like I'll need to see it running to understand all the moving parts -- but that's okay!

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.46-alpha](https://keyman.com/downloads/releases/alpha/17.0.46)

**Pull request merged**